src/riak_cs_quota.erl

```diff
@@ -0,0 +1,207 @@
+%% ---------------------------------------------------------------------
+%%
+%% Copyright (c) 2007-2015 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% ---------------------------------------------------------------------
+
+%% @doc Behaviour for Quota / QoS hook point, to save CS systems by
+%% limiting loads. The idea of Quota and QoS system is, to limit
+%% things by pluggable modules. This is because requirements vastly
+%% varies and it is too hard to cover all of them with single
+%% implementation.
+%% This is experimental feature. These interfaces and behaviousr
+%% may or may not change until general availability.
+%%
+%% To use 'inherited' your own module, it should be configured in
+%% application environment (set it in advanced.config). The item name
+%% is `quota_modules'. Example:
+%%
+%% {riak_cs, [
+%%   {quota_modules, [riak_cs_simple_quota, riak_cs_bw_limitter]},
+%%   ...
+%% ]},
+%%
+%% The order of module names is also the order of evaluation. Modules
+%% are evaluated every time 1) incoming requests passed every other
+%% authentication, authorization works (including ACL, Policy), and 2)
+%% all incoming requests successfully finished even if those requests
+%% resulted in 50x or whatever.
+%%
+%% 1) is a callback called `allow/4' and 2) is `update/2` callback.
+%% These callbacks are called in each process context, 1) is at
+%% mochiweb acceptor and 2) is webmachine log handler.  Other
+%% callbacks are for maintenance or management, which is called at the
+%% context of gen_server process, a child process of
+%% `riak_cs_quota_sup'. Note that these processes can't be added nor
+%% removed at runtime, please just control such enabled/disabled
+%% status within plugin modules.
+%%
+%% Modules may return one of these values as defined in callback:
+%% - ok, will return 20x
+%% - disk quota exceed, returns 507
+%% - bandwidth exceed, returns 429
+%% - access rate exceed, returns 429
+%%
+%% Known limitations:
+%% - currently this is only for objects.
+%% - separate owner and accessor case: bandwidth and access should be
+%%   for accessor, whild owner is for usage
```

Comment:
I want it to be explicitly written because my memory is so bad :smile: 
From source code, for `allow(Owner, Access, Context)`, `Owner` is bucket owner,  `Access` and `Context` have access user (maybe `undefined` for anonymous) on the other hand.